source/extensions/filters/http/transformation/transformer.h

```diff
@@ -48,6 +48,9 @@ class Transformer {
                          const Http::RequestHeaderMap *request_headers,
                          Buffer::Instance &body,
                          Http::StreamFilterCallbacks &callbacks) const PURE;
+  virtual void transform_headers(Http::RequestHeaderMap &,
```

Comment:
instead of 2 methods, can we change transform to be
```  virtual void transform(Http::RequestOrResponseHeaderMap &map,
                         // request header map. this has the request header map
                         // even when transforming responses.
                         const Http::RequestHeaderMap *request_headers,
                         absl::optional<std::reference_wrapper<Buffer::Instance>> body,
                         Http::StreamFilterCallbacks &callbacks) 
```
(i.e. change the body to absl::optional<std::reference_wrapper<Buffer::Instance>>)?
this way the impl of the existing transformers doesn't have to change?

Reply:
The body param wasn't the reason behind the change here. I created this new method because I need to pass in both the request and response maps that can be modified by the transform. Prior to having this second method, I had to make two calls to the `transform` method (once to transform the request headers and then once to transform the response header). It seemed a little simpler to pass in everything I needed at once.

I'm not completely familiar with why we are also passing in a constant `request_header` in `transform` but another change I could make would be to remove the const and pass in the response headers as `map` and then request headers into that param so that I have access to modifying them both with one call. Does that sound better @yuval-k?

Reply:
when this method is called on request, the `request_headers` is nil.
when it is called on the response it is not.
I think we can un-const `request_headers` it. the reason it was passed as const as in the original implementation you never had a reason to change it..